src/lua/loaders.lua

```diff
@@ -331,43 +419,56 @@ local function getenv_boolean(varname, default)
 end
 
 -- true/false if explicitly enabled or disabled, nil otherwise.
-local override_loader_is_enabled
+local override_searcher_is_enabled
 
 -- Whether the override loader is enabled.
-local function override_loader_onoff(_name)
+local function override_searcher_onoff(_name)
     -- Follow the switch if it is explicitly enabled or disabled.
-    if override_loader_is_enabled ~= nil then
-        return override_loader_is_enabled
+    if override_searcher_is_enabled ~= nil then
+        return override_searcher_is_enabled
     end
 
     -- Follow the environment variable otherwise.
     return getenv_boolean('TT_OVERRIDE_BUILTIN', true)
 end
 
-local override_loader = conditional_loader(chain_loaders({
-    prefix_loader('override', package.loaders[2]),
-    prefix_loader('override', package.loaders[3]),
-    prefix_loader('override', package.loaders[4]),
-    prefix_loader('override', package.loaders[5]),
-    prefix_loader('override', package.loaders[6]),
-    prefix_loader('override', package.loaders[7]),
-}), override_loader_onoff)
+rawset(searchers, 'override', conditional_searcher(chain_searchers({
+    prefix_searcher('override', searchers[2]),
+    prefix_searcher('override', searchers[3]),
+    prefix_searcher('override', searchers[4]),
+    prefix_searcher('override', searchers[5]),
+    prefix_searcher('override', searchers[6]),
+    prefix_searcher('override', searchers[7]),
+}), override_searcher_onoff))
 
--- Add two loaders:
+-- Join three searchers:
 --
--- - Search for override.<module_name> module. It is necessary for
---   overriding built-in modules.
--- - Search for a built-in module (compiled into tarantool's
+-- - Searcher for preload LuaJIT modules.
+-- - Searcher for override.<module_name> module. It is necessary
+--   for overriding built-in modules.
+-- - Searcher for a built-in module (compiled into tarantool's
 --   executable).
 --
--- Those two loaders are mixed into the first loader to don't
--- change ordinals of the loaders 2-8. It is possible that someone
--- has a logic based on those loader positions.
-package.loaders[1] = chain_loaders({
-    package.loaders[1],
-    override_loader,
-    builtin_loader,
-})
+-- Those three searchers are mixed into the first searcher to
+-- don't change ordinals of the searchers (and hence, loaders)
+-- 2-8. It is possible that someone has a logic based on those
+-- loader positions.
+rawset(searchers, 1, chain_searchers({
+    searchers['preload'],
+    searchers['override'],
+    searchers['builtin'],
+}))
+
+-- XXX: The default LuaJIT loaders are:
+-- [1]: preload
+-- [2]: package.path
+-- [3]: package.cpath
+-- [4]: croot
+--
+-- Tarantool loaders just maps the <searchers> table.
+for index, _ in ipairs(searchers) do
+    rawset(package.loaders, index, gen_legacy_loader(searchers, index))
+end
```

Comment:
Maybe it worth to highlight that tarantool historically has different loader ordinals and they're preserved here (IOW, the default LuaJIT loader ordinals are NOT preserved).

Feel free to ignore.

Reply:
OK, added.
```diff
diff --git a/src/lua/loaders.lua b/src/lua/loaders.lua
index bdc1bd4d72..c175a1b843 100644
--- a/src/lua/loaders.lua
+++ b/src/lua/loaders.lua
@@ -465,7 +465,15 @@ rawset(searchers, 1, chain_searchers({
 -- [3]: package.cpath
 -- [4]: croot
 --
--- Tarantool loaders just maps the <searchers> table.
+-- Tarantool loaders just maps the <searchers> table:
+-- [1]: preload -> override -> builtin
+-- [2]: path.cwd.dot -> [path.app.dot]
+-- [3]: cpath.cwd.dot -> [cpath.app.dot]
+-- [4]: path.cwd.rocks -> [path.app.rocks]
+-- [5]: cpath.cwd.rocks -> [cpath.app.rocks]
+-- [6]: package.path
+-- [7]: package.cpath
+-- [8]: croot
 for index, _ in ipairs(searchers) do
     rawset(package.loaders, index, gen_legacy_loader(searchers, index))
 end
```